README.md

```diff
@@ -1,9 +1,39 @@
 # CSI Proxy
 
-CSI Proxy is a binary that exposes a set of gRPC APIs around storage operations
-over named pipes in Windows. A container, such as CSI node plugins, can mount
-the named pipes depending on operations it wants to exercise on the host and
-invoke the APIs.
+CSI Proxy (which might be more aptly named "csi-node-proxy") is a binary that exposes a set of gRPC APIs around local storage operations for nodes in Windows. A container, such as CSI node plugins, can mount the named pipes depending on operations it wants to exercise on the host and
+invoke the APIs.  This allows a storage plugin to run as if were a CSI plugin on linux which had the ability to do "priveleged" actions on the windows host OS.
+
+```
+                                              +------------------+
++---------------------------------------------|                  |
+|                 +----------+                |                  |
+|                 |csi node  |                |   CSI Node Plugin|
+|                 |driver    +---------------->                  |
+|                 |registrar |                |                  |
+| NodeStageVolume ----^------+                ----------+--------+
+| NodeUnstageVol      |                                 |  Disk APIs
+| NodePublishVol      |GetNodeInfo                      |  Volume APIs
+| NodeResizeVol       |NodeRegistrationStatus    /pipe  |  SMB APIs
+| NodeGetInfo         |                                 |  FileSystem/System APIs
+| NodeGetVolStats     | csi-plugin-reg.sock   +---------v---------+
+|                     |                       |                   |
+|                 + --+--------+              |    csi-proxy.exe  |
+|                 | kubelet.exe|              +-------------------+
+|                 |            |                      |
+|                 +---+--------++                     |
+|                     |                               |
+| csi.sock            |                               |
+|                     v                               v
+|                 +---+-------------------------------+------+
+|                 |                                          |
++---------------->|         Host Os (Windows)                |
+                  +------------------------------------------++
+```
+In the above diagram, there are 3 communication channels being utilized for CSI proxy:
+
+- `csi.sock` : This lives at /var/lib/kubelet/plugins/csi-plugin/csi.sock, and is used for the host OS to communicate with the CSI node plugin
```

Comment:
Typically the path names in Windows (assuming OS installed in C:\ in host) would be `C:\var\lib\kubelet`. You may also want to add the qualifier "This usually lives ..." as the installation paths can be customized by the distribution of Kubernetes.

Reply:
done